### ISSUE TITLE
Support for MR via fork

### ISSUE THREAD

**Issue by eperdeme** [samcontesse/gitlab-merge-request-resource#13]
Hello,

We're trying to use this resource but it looks like it does not support a MR when source is a Fork.

Does anybody have the code to do this stashed away somewhere ?

**Comment by samcontesse** [comment]
Hello @eperdeme, I've never tried with forked repositories. Do you have more information about the error ? What error is return from the API ?

**Comment by eperdeme** [comment]
So the issue is the clone needs to use the forks repo to pull in the code, the other older gitlab resource had the same issue but the project looks stale; https://github.com/swisscom/gitlab-merge-request-resource/pull/31

The errors
```
Cloning into '/tmp/build/get'...
error merging 18e846d87944f2a40767eef0dbf43f66d1c35d24 into develop: exit status 1
````

**Comment by samcontesse** [comment]
@eperdeme would you mind to test the following image ? 

```
- name: merge-request
  type: docker-image
  source:
    repository: samcontesse/gitlab-merge-request-resource
    tag: feature_add-fork-repository-support
```

Let me know if it works with your configuration.

**Comment by eperdeme** [comment]
Looks to be working great on the existing job I've just run. Thank you!

**Comment by eperdeme** [comment]
Oh it does error;

2019/04/26 14:21:07 Set commit status failed: 404, response 
```
- name: git
  type: merge-request
  tags: [dev]
  source:
    uri: https://git.nonprod.williamhill.plc/tf-community-modules/tf-aws-eks.git
    private_token: ((gitlab_api_token))
    insecure: true

- name: terraform-validate
  plan:
  - get: git
    trigger: true
    tags: [dev]
  - get: scripts
    tags: [dev]
  - task: validate-code
    file: scripts/validate.yml
    input_mapping:
      product: git
    tags: [dev]
  - put: git
    params:
      repository: product
    tags: [dev]
  on_failure:
    put: git
    tags: [dev]
    params:
      repository: git
      status: failed
  on_success:
    put: git
    tags: [dev]
    params:
      repository: git
      status: success
      comment:
        text: |
          Terraform validate Passed.
```

Does the git resource still know the MR repo url for submitting sucess/failures

**Comment by samcontesse** [comment]
I fixed it in 6ce01091 but it will also depend on how permissions are granted. There are a few open issues on GitLab side regarding fork/target commit statuses :

* https://gitlab.com/gitlab-org/gitlab-ce/issues/22238
* https://gitlab.com/gitlab-org/gitlab-ce/issues/59649

The new image (tag `feature_add-fork-repository-support`) is building and will be available soon. Feedback welcome ;-)

**Comment by eperdeme** [comment]
Morning, Just tested and looks to work as intended now.  Time to start swapping out Jenkins with Conrouse.

Really shame on that outstanding bug with gitlab, we shall have to work around it as best we can, luckily our gitlabs internal so maybe we extend the group permissions for CI users.
